pkg/network/filtermanager.go

```diff
@@ -69,48 +69,43 @@ func (fm *filterManager) InitializeReadFilters() bool {
 	if len(fm.upstreamFilters) == 0 {
 		return false
 	}
-
-	fm.onContinueReading(nil)
-	return true
-}
-
-func (fm *filterManager) onContinueReading(filter *activeReadFilter) {
 	var index int
 	var uf *activeReadFilter
-
-	if filter != nil {
-		index = filter.index + 1
-	}
-
+	var status api.FilterStatus
 	for ; index < len(fm.upstreamFilters); index++ {
 		uf = fm.upstreamFilters[index]
 		uf.index = index
 
 		if !uf.initialized {
 			uf.initialized = true
-
-			status := uf.filter.OnNewConnection()
-
+			status = uf.filter.OnNewConnection()
 			if status == api.Stop {
-				return
+				return false
 			}
 		}
+	}
+	return true
+}
 
-		buf := fm.conn.GetReadBuffer()
+func (fm *filterManager) onContinueReading(previousIndex int, buf buffer.IoBuffer) {
+	var index int
+	var uf *activeReadFilter
 
-		if buf != nil && buf.Len() > 0 {
-			status := uf.filter.OnData(buf)
+	if previousIndex != 0 {
+		index = previousIndex + 1
+	}
 
-			if status == api.Stop {
-				//fm.conn.Write("your data")
-				return
-			}
+	for ; index < len(fm.upstreamFilters); index++ {
```

Comment:
I'm not sure if it is safe to skip checking `uf.initialized` here.
@taoyuanyuan @nejisama may confirm it.